datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/DataGenEntry.scala

```diff
@@ -0,0 +1,110 @@
+package com.nvidia.rapids.tests.scaletest
+
+import org.apache.spark.sql.SparkSession
+import scopt.OptionParser
+
+object DataGenEntry{
+
+  // case class represents all input commandline arguments
+  private case class Config(scaleFactor: Int = 1,
+    complexity: Int = 1,
+    tables: Array[String] = Array(),
+    outputDir: String = "",
+    format: String = "parquet",
+    version: String = "")
+
+  private def runDataGen(config: Config): Unit ={
+    // Init SparkSession
+    val spark = SparkSession.builder()
+      .appName("Scale Test Data Generation")
+      .getOrCreate()
+
+    val tableGenerator = new TableGenerator(config.scaleFactor, config.complexity, spark)
+    val tableMap = tableGenerator.genTables(config.tables)
+    val baseOutputPath = s"${config.outputDir}/" +
+      s"SCALE_" +
+      s"${config.scaleFactor}_" +
+      s"${config.complexity}_" +
+      s"${config.format}_" +
+      s"${config.version}"
+    // 512 Mb block size requirement
+    val setBlockSizeTables = Seq("b_data", "g_data")
+
+    for ((tableName, df) <- tableMap) {
+      if (setBlockSizeTables.contains(tableName)) {
+        config.format match {
+          case "parquet" => df.write
```

Comment:
Can we either make this an enum/sealed trait or add in a default case so we can catch errros without relying on scala to insert one for us.

Reply:
Add one for unsupported to throw exception and shorten the code.